bip-0360.mediawiki

```diff
@@ -176,137 +177,196 @@ It is proposed to use SegWit version 3. This results in addresses that start wit
+provided strong evidence that this was not a feasible approach for the NIST approved Post-Quantum signature algorithms.
+It is an open question if Post-Quantum signature algorithms can be designed to provide JPEG resistance.
+
+==== Future Considerations ====
+
+While P2QRH lacks features like signature aggregation, it offers a pragmatic first step
+toward quantum resistance. Future BIPs can add enhancements like signature aggregation, and possibly full
 BIP-32 compatibility once tested and viable. Until quantum cryptography hardware and advanced schemes are widespread,
-P2QRH provides meaningful protection against quantum threats without delaying deployment for a perfect solution.
+P2QRH provides meaningful protection against quantum threats as early as possible.
 
-Additional follow-on BIPs will be needed to implement P2QS, signature aggregation, and full BIP-32 compatibility
+Additional follow-on BIPs will be needed to implement signature aggregation, and full BIP-32 compatibility
 (if possible) <ref name="bip-32">BIP-32 relies on elliptic curve operations to derive keys from xpubs to support
 watch-only wallets, which PQC schemes may not support.</ref>. However, until specialized quantum cryptography hardware
-is widespread and signature aggregation schemes are thoroughly vetted, P2QRH addresses should be an adequate
-intermediate solution that provides meaningful protection against quantum threats.
+is widespread and signature aggregation schemes are thoroughly vetted, P2QRH addresses are an intermediate solution 
+to quantum threats.
 
 == Specification ==
 
 We define the signature scheme and transaction structure as follows.
 
-=== Descriptor Format ===
+=== Pay to Quantum Resistant Hash (P2QRH) ===
 
-To integrate P2QRH into existing wallet software and scripts, we introduce a new output descriptor function
-<code>qrh()</code>. This function represents a P2QRH output, similar to how <code>wpkh()</code> and <code>tr()</code>
-are used for P2WPKH and P2TR outputs, respectively.
+A P2QRH output is simply the root of the tapleaf merkle tree defined as in [https://github.com/bitcoin/bips/blob/master/bip-0341.mediawiki BIP-341] 
+and used as an internal value in P2TR.
 
-The <code>qrh()</code> function takes a threshold value and multiple key specifications grouped by key type. The format is:
+To construct a P2QRH output we follow the same process as [https://github.com/bitcoin/bips/blob/master/bip-0341.mediawiki BIP-341]
+to compute the tapscript merkle root. However, instead of the root of the Merkle tree being hashed together with the internal
+key in P2QRH the root is hashed by itself using the tag "QuantumRoot".
 
-  qrh(threshold, keytype(0x01, [hash1, hash2, ...]), keytype(0x02, [hash1, hash2, ...]), ...)
+<source>
+D = tagged_hash("TapLeaf", bytes([leaf_version]) + ser_script(script))
```

Comment:
```suggestion
A..E = tagged_hash("TapLeaf", bytes([leaf_version]) + ser_script(script))
```

Reply:
In  [BIP-0341](https://github.com/bitcoin/bips/blob/master/bip-0341.mediawiki) it is given the same way using D

```
<source>
D = tagged_hash("TapLeaf", bytes([leaf_version]) + ser_script(script))
CD = tagged_hash("TapBranch", C + D)
CDE = tagged_hash("TapBranch", E + CD)
ABCDE = tagged_hash("TapBranch", AB + CDE)
TapTweak = tagged_hash("TapTweak", p + ABCDE)
</source>
```

I agree your change would be an improvement on it's own would be confusing when compared with BIP-341 since it might imply we are doing something different with the leafs.